app/src/main/java/com/jbc7ag/cryptso/ui/currencydetail/CurrencyDetailFragment.kt

```diff
@@ -0,0 +1,110 @@
+package com.jbc7ag.cryptso.ui.currencydetail
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.core.content.ContextCompat
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import com.bumptech.glide.Glide
+import com.bumptech.glide.request.RequestOptions
+import com.jbc7ag.cryptso.R
+import com.jbc7ag.cryptso.data.model.BookDetail
+import com.jbc7ag.cryptso.data.model.OrderDetail
+import com.jbc7ag.cryptso.databinding.FragmentCurrencyDetailsBinding
+import com.jbc7ag.cryptso.util.*
+import dagger.hilt.android.AndroidEntryPoint
+
+enum class LISTTYPE{
+    BIDS, ASKS
+}
+@AndroidEntryPoint
+class CurrencyDetailFragment: Fragment() {
+
+    private lateinit var binding: FragmentCurrencyDetailsBinding
+    private val viewModel: CurrencyViewModel by viewModels()
+    private lateinit var bidsAdapter: BidsAdapter
+
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        return FragmentCurrencyDetailsBinding.inflate(layoutInflater, container, false)
+            .apply { binding = this }
+            .root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        val bookName = requireArguments().getString("currency_id")
```

Comment:
We can use the safeArgs generated class :)
```suggestion
       val args = CurrencyDetailFragmentArgs.fromBundle(requireArguments())
       val bookName = args.currencyId
```